### ISSUE TITLE
DBT support

### ISSUE THREAD

**Issue by sivankumar86** [jupyterlab/jupyterlab#13529]
We wants web based DBT IDE and it would be great if we can added DBT support in jupyter lab

**Comment by sivankumar86** [comment]
@krassowski thanks for taking a look. dbt is a tool to develop models in datalake such as snowflake ,databricks etc.. Find more detail here 

https://docs.getdbt.com/docs/introduction

**Comment by andrii-i** [comment]
@sivankumar86 what do you need jupyterlab to provide? Is this something you could add as an extension?

**Comment by sivankumar86** [comment]
Yes, something like extension but, i am not familiar with extension development.

**Comment by jweill-aws** [comment]
It's unclear what you're asking for here. Are you looking for a custom kernel that can execute dbt? An extension that adds functionality related to dbt? What are you expecting out of JupyterLab that the [existing web-based DBT IDE](https://www.getdbt.com/blog/announcing-the-dbt-ide/) doesn't already provide?

**Comment by sivankumar86** [comment]
@jweill-aws Thanks for reply, [web-based DBT IDE] is not open source one. I am looking for similar one. High level, I should be able to edit dbt code in jupyter lab and I  should be able to compile, run and debug the code hence, it can be extension or cell magic.

**Comment by jweill-aws** [comment]
OK, thanks. I suspect that a kernel is best suited for this. See the documentation on kernels here: https://docs.jupyter.org/en/latest/projects/kernels.html

**Comment by sivankumar86** [comment]
@jweill-aws Thank you for reply. Editor extension would be great instead of kernel. Example, it should show lineage for dbt sql file and should highlight the sql syntax .

**Comment by bollwyvl** [comment]
You could probably run [dbt-language-server](https://github.com/fivetran/dbt-language-server) under [jupyterlab-lsp](https://github.com/jupyter-lsp/jupyterlab-lsp) to get part of the way there.

But I don't seem like a broadly-enough-applicable problem to get any special treatment in core jupyterlab.
